instrumentation/resources/library/src/test/java/io/opentelemetry/instrumentation/resources/CgroupV2ContainerIdExtractorTest.java

```diff
@@ -33,34 +33,58 @@ void fileNotReadable() {
     assertThat(result).isSameAs(Optional.empty());
   }
 
-  @Test
-  void extractSuccess_docker() throws Exception {
+  @SuppressWarnings("unchecked")
+  private void verifyContainerId(String rawFileContent, String containerId) throws Exception {
     when(filesystem.isReadable(V2_CGROUP_PATH)).thenReturn(true);
-    Stream<String> fileContent = getTestDockerFileContent();
-    when(filesystem.lines(V2_CGROUP_PATH)).thenReturn(fileContent);
+    Stream<String> fileContent = fileToStreamOfLines(rawFileContent);
+    Stream<String> fileContent1 = fileToStreamOfLines(rawFileContent);
+    when(filesystem.lines(V2_CGROUP_PATH)).thenReturn(fileContent, fileContent1);
     CgroupV2ContainerIdExtractor extractor = new CgroupV2ContainerIdExtractor(filesystem);
     Optional<String> result = extractor.extractContainerId();
-    assertThat(result.orElse("fail"))
-        .isEqualTo("be522444b60caf2d3934b8b24b916a8a314f4b68d4595aa419874657e8d103f2");
+    assertThat(result.orElse("fail")).isEqualTo(containerId);
+  }
+
+  @Test
+  void extractSuccess_docker() throws Exception {
+    verifyContainerId(
+        "docker_proc_self_mountinfo",
+        "be522444b60caf2d3934b8b24b916a8a314f4b68d4595aa419874657e8d103f2");
+  }
+
+  @Test
+  void extractSuccess_docker1() throws Exception {
+    verifyContainerId(
+        "docker_proc_self_mountinfo1",
+        "188329f95b930c32eeeffd34658ed2538960947e166743fa3743f5ce3d739b40");
+  }
+
+  @SuppressWarnings("unchecked")
```

Comment:
is this `@SuppressWarnings` really needed?

Reply:
removed.